os/kernel/pthread/pthread_keycreate.c

```diff
@@ -91,65 +92,40 @@
 
 int pthread_key_create(FAR pthread_key_t *key, pthread_destructor_t destructor)
 {
-	struct pthread_tcb_s *rtcb = (struct pthread_tcb_s *)this_task();
-	struct task_group_s *group = rtcb->cmn.group;
-	struct pthread_key_s *new_key;
-	struct pthread_key_s *cur_key;
-	pthread_key_t new_key_inex = 1;
+	struct tcb_s *rtcb = this_task();
```

Comment:
you are accessing group as follows in pthread_getspecific() , but inside pthread_key_create, accessing it directly from tcb_s data type. Is there any specific reason ? 

pthread_getspecific() {
struct pthread_tcb_s *rtcb = (FAR struct pthread_tcb_s *)this_task(); 
struct task_group_s *group = rtcb->cmn.group;
}

Reply:
When I need to use pthread tcb and group, I used getspecific's method. But when I don't need to use pthread tcb, I used key creation's method.
key data is in pthread tcb, and
key destructor and flag is in group.